spec/lib/linters/enhanced_idv_events_linter_spec.rb

```diff
@@ -0,0 +1,313 @@
+require 'rubocop'
+require 'rubocop/rspec/cop_helper'
+require 'rubocop/rspec/expect_offense'
+
+require_relative '../../../lib/linters/enhanced_idv_events_linter'
+
+RSpec.describe RuboCop::Cop::IdentityIdp::EnhancedIdvEventsLinter do
+  include CopHelper
+  include RuboCop::RSpec::ExpectOffense
+
+  let(:config) { RuboCop::Config.new }
+  let(:cop) { RuboCop::Cop::IdentityIdp::EnhancedIdvEventsLinter.new(config) }
+  let(:check_param_docs) { false }
+
+  before do
+    if !check_param_docs
+      # Most of these tests don't involve the @param doc linter, so
+      # unwire it here.
+      allow(cop).to receive(:check_arg_has_docs).and_return(nil)
+    end
+  end
+
+  it 'does not register an offense for non-idv methods' do
+    expect_no_offenses(<<~RUBY)
+      module AnalyticsEvents
+        def a_non_idv_method
+            track_event(:a_non_idv_method)
+        end
```

Comment:
unsure if we care but this is an indent by 4 instead of our usual by 2
```suggestion
        def a_non_idv_method
          track_event(:a_non_idv_method)
        end
```